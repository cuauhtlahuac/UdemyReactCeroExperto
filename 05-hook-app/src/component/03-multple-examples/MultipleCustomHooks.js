import React from 'react'
import useFetch from 'hooks/useFetch'
import useCounter from 'hooks/useCounter'

const MultipleCustomHooks = () => {

    const { counter, increment } = useCounter(1)

    const url = `https://www.breakingbadapi.com/api/quotes/${ counter }`

    const {
        data,
        loading,
    } = useFetch( url );

    const { author, quote } = !!data && data[0];

    return (
        <>
          <h1>Breaking Bad Quotes</h1>
            <hr/>

            { loading ?
            (
                <div className="d-flex mb-2 justify-content-end">
                    <div className="spinner-grow bg-info" style={{ width: '3rem', height: '3rem' }} role="status">
                        <span className="sr-only">Loading...</span>
                    </div>
                </div>
            ) : (
                <blockquote className="blockquote text-right">
                    <p className="mb-0">{ quote }</p>
                    <footer className="blockquote-footer">{ author }</footer>
                </blockquote>
            )}

            <button className="btn btn-primary" onClick={ increment }>
                next quote 
            </button>
        </>
    )
}

export default MultipleCustomHooks
