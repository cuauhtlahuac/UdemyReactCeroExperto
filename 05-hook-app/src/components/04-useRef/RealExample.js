import React, { useState } from 'react'
import MultipleCustomHooks from 'components/03-multple-examples/MultipleCustomHooks'

const RealExample = () => {

    const [show, setShow] = useState( false )

    return (
        <div>
            <h1>Use Ref Real Example</h1>
            <hr />

            { show && <MultipleCustomHooks /> } 

            <br />
            <br />
            <button className="btn btn-primary" onClick={ ()=> setShow( !show )}>
                { show ? "Hide" : "Show" }
            </button>
        </div>
    )
}

export default RealExample
