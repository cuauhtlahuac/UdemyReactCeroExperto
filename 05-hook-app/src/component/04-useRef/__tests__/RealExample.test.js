import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom';

import RealExample from '../RealExample';
import MultipleCustomHooks from 'component/03-multple-examples/MultipleCustomHooks';

describe('Real Example Tests', () => {
	let component;
	beforeEach(() => {
		component = mount(<RealExample />);
	});

	afterEach(() => {
		component.unmount();
	});

	test('should show correctly', () => {
		expect(component).toMatchSnapshot();
	});

	test('should change the button text', () => {
		const button = component.find('button');

		expect(button.text()).toBe('Show');

		act(() => {
			button.props().onClick();
    });
    
		expect(button.text()).toBe('Hide');
	});

	test('should show <MultipleCustomHooks />', async () => {
		const button = component.find('button');
		expect(component.find('MultipleCustomHooks').exists()).toBe(false);

		button.simulate('click');

		expect(component.find('MultipleCustomHooks').exists()).toBe(true);
	});
});
