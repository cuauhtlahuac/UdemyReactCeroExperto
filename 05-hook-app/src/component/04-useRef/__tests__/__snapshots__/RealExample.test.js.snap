// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Real Example Tests should show <MultipleCustomHooks /> 1`] = `
<div>
  <h1>
    Use Ref Real Example
  </h1>
  <hr />
  <br />
  <br />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show
  </button>
</div>
`;

exports[`Real Example Tests should show correctly 1`] = `
<div>
  <h1>
    Use Ref Real Example
  </h1>
  <hr />
  <br />
  <br />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show
  </button>
</div>
`;
